api/types/events/api.go

```diff
@@ -71,10 +71,24 @@ type AuditEvent interface {
 	SetClusterName(string)
 }
 
+type AuditContext struct {
+	context.Context
```

Comment:
I would consider making this a private field and not an embed. This pushes callers to the constructor, meaning we more reliably strip the cancelations (plus it forces the design to be more explicit about the API).

If we do need to provide direct access to the underlying context, we can do so via a separate method, but so far I'm not sure why would want that.

Reply:
This access the underlying context, however I guess this shouldnt be needed now, right? https://github.com/gravitational/teleport/blob/lxea%2Faudit-context/lib/events/emitter.go#L113-L114